### PR TITLE
Fix StatusBar (React Native < 20)

### PR DESCRIPTION
**Pull request by LeoLeBras**

### CODE REVIEW

**Comment by davidLeonardi** [comment]
Thanks for the contribution.

Not sure i like the idea of adding fixes for old versions, now that 0.21 is out and 0.22 is about to be released.
IMO if you're on old versions of RN just deal with the breakage, and upgrade to get stuff fixed.
Perhaps i'm too strict though.
@charpeni , opinions?

File: index.js [hunk 1]
```diff
@@ -379,6 +379,21 @@ class Router extends React.Component {
         }
       }
 
+      // StatusBar isn't available
+      else {
```
Comment (charpeni) [inline index.js:383]: Closing curly brace does not appear on the same line as the subsequent block.

File: index.js [hunk 2]
```diff
@@ -379,6 +379,21 @@ class Router extends React.Component {
...
+        if (Platform.OS === 'ios') {
+          if (this.props.statusBarProps) {
+            statusBarProps = this.props.statusBarProps;
+          }
+          if (this.state.route && this.state.route.statusBarProps) {
+            statusBarProps = _.defaults(this.state.route.statusBarProps, statusBarProps);
+          }
+          if(statusBarProps && statusBarProps.barStyle) {
```
Comment (charpeni) [inline index.js:391]: Keyboard "if" must be followed by whitespace.

File: index.js [hunk 3]
```diff
@@ -379,6 +379,21 @@ class Router extends React.Component {
...
+            React.StatusBarIOS.setStyle(statusBarProps.barStyle)
```
Comment (charpeni) [inline index.js:392]: Missing semicolon.

**Comment by charpeni** [comment]
I think you're right @sethx. 

Especially considering that `StatusBarIOS` is an API and has a behavior that differs from `StatusBar` component. I think `StatusBarIOS` should be handle on app side instead of react-native-simple-router.

Thank you for the contribution, I'll close this pull request since we believe that it's not viable.

**Pull request closed without merging**